### PR TITLE
feat: colors foundation page

### PR DESCRIPTION
**Pull request by shleewhite**
- [x] Create the page
- [x] Add page to sidebar
- [x] Create new ColorSwatch component that displays the color
- [x] Create new ColorGradientRainbow that displays all the color aliases from darkest -> lightest
- [x] Export colorRed, colorOrange, colorYellow, colorGreen, colorBlue, colorPurple aliases from design-tokens 

**Contributing to Twilio**

> All third-party contributors acknowledge that any contributions they provide will be made under the same open-source license that the open-source project is provided under.

- [ ] I acknowledge that all my contributions will be made under the project's license.

### CODE REVIEW

File: packages/paste-website/src/components/color-swatch/ColorGradient.tsx [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+const aliasPrefixes = ['colorGray', 'colorRed', 'colorOrange', 'colorYellow', 'colorGreen', 'colorBlue', 'colorPurple'];
+
+type AliasName = keyof typeof colors;
+
+const sortAliasNames = (aliasNames: string[]): string[] => {
+  return aliasNames.sort((nameA, nameB) => nameA.localeCompare(nameB, undefined, {numeric: true})).reverse();
+};
+
+const getAliasValuesFromPrefix = (prefix: string, theme: 'default' | 'dark'): string[] => {
+  const themeColors = theme === 'default' ? colors : darkColors;
```
Comment (shleewhite) [inline packages/paste-website/src/components/color-swatch/ColorGradient.tsx:17]: Tried to future proof it so if the dark palette colors change the gradient will look correct.

**Comment by shleewhite** [comment]
There are known bugs with a couple of the tokens displayed:
Screen Shot 2022-02-08 at 5 31 24 PM
Screen Shot 2022-02-08 at 5 31 34 PM

These will be fixed in [the dark mode PR](https://github.com/twilio-labs/paste/pull/2184#pullrequestreview-876776722).

**Comment by shleewhite** [comment]
The images also look a little blurry... I exported them from figma, is that not the right way?
Screen Shot 2022-02-08 at 5 33 09 PM

File: packages/paste-design-tokens/tokens/global/palette-color.yml [hunk 2]
```diff
@@ -38,6 +38,186 @@ props:
   color-gray-100:
     value: "{!palette-gray-100}"
     comment: "Gray Color 10"
+  color_red_1:
```
Comment (SiTaggart) [inline packages/paste-design-tokens/tokens/global/palette-color.yml:41]: OK so, aliases should not be added to the theme. When you create or add tokens, you will automatically export them as a public API on the design token package, and then automatically add them to the React theme context in the Theme Package. Effectively adding them as a public API to the theme package too. Then, depending on the type of token you create, you might automatically add it to Box style props, and it's typings. (The color bucket is the only bucket of tokens that is not assigned to a style props right now, so this specific example isn't exposed automatically at the Box theme).

What you are doing in this file is actually creating Design Tokens. What you want to do instead is make sure to add any missing _aliases_ to the color palette aliases. https://github.com/twilio-labs/paste/blob/main/packages/paste-design-tokens/tokens/aliases/color-palette.yml

These are then available in https://unpkg.com/browse/@twilio-paste/design-tokens@7.0.0/dist/tokens.raw.json which is exported by the design tokens package.

You could probably then add that to the gatsby graphql if you like, but importing that json file from design tokens is probably just enough.

`import {aliases} from '@twilio-paste/design-tokens/dist/tokens.raw.json'`
`import {aliases} from '@twilio-paste/design-tokens//dist/themes/dark/tokens.raw.json'`

File: packages/paste-website/src/components/color-swatch/ColorGradient.tsx [hunk 3]
```diff
@@ -0,0 +1,60 @@
+import * as React from 'react';
+import {styled, themeGet} from '@twilio-paste/styling-library';
+import {colors} from '@twilio-paste/design-tokens';
+import {colors as darkColors} from '@twilio-paste/design-tokens/dist/themes/dark/tokens.es6';
```
Comment (SiTaggart) [inline packages/paste-website/src/components/color-swatch/ColorGradient.tsx:4]: So these two are the imports that should be 

```
import {aliases} from '@twilio-paste/design-tokens/dist/tokens.raw.json'
import {aliases} from '@twilio-paste/design-tokens/dist/themes/dark/tokens.raw.json'
```

File: packages/paste-design-tokens/tokens/global/palette-color.yml (hunk 2 above)
Comment (shleewhite) [reply packages/paste-design-tokens/tokens/global/palette-color.yml:41]: Ohhhhhhhhhhhhhhhhhhhhhh ok. These files are all very similar so i guessed where to add them 😬

**Review by TheSisb — APPROVED** [review]
Super clean 👍

**Review by SiTaggart — COMMENTED** [review]
Implementation is great. 

It would be nice to have some stories for the custom components so we get some VRT of the custom components. There are some filtering functions that might need units tests, but I could go either way. Bonus points for a Cypress applitools test to ensure page VRT.

**Comment by katieporter** [comment]
Daaaaang this is looking good! A couple of the tiniest things:

- Can the in-page nav sit _below_ the header, like so? 
image
- Is it possible to add a non-linked breadcrumb to the header that says "Foundations", like so? (Ideally we'd do this for all pages within Foundations but we can do that later)
image

And a lil scope creep, hopefully not too bad:
- In the text colors section, can we swap `color-text-icon-inverse` with just `color-text-inverse` and have it sit on top of `color-background-body-inverse`?
- In the background colors section, could we drop `color-background-strongest` and add `color-background-body` as the first token in the section? It'll need a border around it - probably `color-border-weaker`.

File: packages/paste-website/src/component-examples/ColorsFoundationExamples.tsx [hunk 4]
```diff
@@ -0,0 +1,277 @@
...
+export const InteractingStatesImgCaption: React.FC = () => (
+  <ImgCaption>
+    In this example, we are using a combination of color and shadow to help users identify the primary button while in
+    it’s different states.
+  </ImgCaption>
+);
+
+export const UseOfColorImgCaption: React.FC = () => (
+  <ImgCaption>
+    In this example, a success toast uses multiple distinguishing visual elements– the thick left border and the icon–
```
Comment (katieporter) [inline packages/paste-website/src/component-examples/ColorsFoundationExamples.tsx:241]:
```suggestion
    In this example, a success toast uses multiple distinguishing visual elements—the thick left border and the icon—
```

File: packages/paste-website/src/component-examples/ColorsFoundationExamples.tsx [hunk 5]
```diff
@@ -0,0 +1,277 @@
...
+    tokens reference an alias. In this example, the $color-background-primary and $color-text-link tokens reference the
+    $color-palette-blue-60 alias to define the color of the token. However, each token is applied to a specific need for
+    different components, such as the background color of a primary button or the color of the anchor.
+  </ImgCaption>
+);
+
+export const InteractingStatesImgCaption: React.FC = () => (
+  <ImgCaption>
+    In this example, we are using a combination of color and shadow to help users identify the primary button while in
+    it’s different states.
```
Comment (katieporter) [inline packages/paste-website/src/component-examples/ColorsFoundationExamples.tsx:235]:
```suggestion
    its different states.
```

File: packages/paste-website/src/pages/foundations/colors/index.mdx [hunk 6]
```diff
@@ -0,0 +1,358 @@
...
+
+If you’re looking to create a new or custom theme, [check out our theme documentation](/customization/creating-a-custom-theme).
+
+#### Default theme
+
+Paste, by default, uses the design language that was specifically designed to unify all Twilio products, known as the Default Theme. It’s currently the standard for almost all Twilio products.
+
+#### Dark theme
+
+Dark Theme is in “early preview” and more work will be done to ensure color contrasts are accessible. Eventually, Dark Theme will be available for wider adoption, but this relies on full Paste adoption in a product. If there are parts of your UI that are not built with Paste, you should not adopt this early preview theme. If there are parts of the UI that aren’t using Paste, Dark theme wouldn’t be applied and the result will be a broken and inconsistent UI. For more questions, [join our discussion on GitHub](https://github.com/twilio-labs/paste/discussions/1527).
```
Comment (katieporter) [inline packages/paste-website/src/pages/foundations/colors/index.mdx:226]:
```suggestion
Paste also has a Dark theme that can be leveraged to enable dark mode within products. This, however, relies on full Paste adoption in a product. If there are parts of the UI that aren’t using Paste, Dark theme wouldn’t be applied and the result will be a broken and inconsistent UI. For more questions, [join our discussion on GitHub](https://github.com/twilio-labs/paste/discussions/1527).
```

File: packages/paste-website/src/pages/foundations/colors/index.mdx [hunk 7]
```diff
@@ -0,0 +1,358 @@
...
+
+Status text and icon color tokens are designed to be accessible on the standard background color tokens and on the weakest variant of the corresponding status background color token. For example, `$color-text-success` can be used on top of `$color-background-success-weakest`. `$color-text` can also be used on the weakest variant of all status background color tokens.
+
+<StatusPairingSwatches />
+
+The inverse text and icon color tokens (`$color-text-inverse`, `$color-text-inverse-weak`, `$color-text-icon-inverse`) are meant to be used on the inverse background color tokens (`$color-background-inverse` and `$color-background-body-inverse`).
+
+<InversePairingSwatches />
+
+A comprehensive list of all accessible color pairings is coming soon!
```
Comment (katieporter) [inline packages/paste-website/src/pages/foundations/colors/index.mdx:178]:
```suggestion

```

**Review by SiTaggart — APPROVED** [review]

**Pull request merged**